test/model/unreads_test.dart

```diff
@@ -469,6 +469,192 @@ void main() {
+          origMessages: unreadMessages,
+          newStreamId: newChannel.streamId,
+          newTopicStr: newTopic));
+        checkNotifiedOnce();
+        checkMatchesMessages(copyMessagesWith(unreadMessages,
+          newChannel: newChannel, newTopic: newTopic));
+        final newMessageIds =
+          model.streams[newChannel.streamId]![TopicName(newTopic)]!;
+        check(originalMessageIds).identicalTo(newMessageIds);
+      });
+
+      test('moving some read messages from a conversation', () async {
+        await prepareStore();
+        final messagesToMove = readMessages.take(2).toList();
+        fillWithMessages(unreadMessages + readMessages);
+
+        model.handleUpdateMessageEvent(eg.updateMessageEventMoveFrom(
+          origMessages: messagesToMove,
+          newTopicStr: newTopic));
+        checkNotNotified();
+        checkMatchesMessages(unreadMessages);
+      });
+
+      test('moving some unread messages from a conversation', () async {
+        await prepareStore();
+        final messagesToMove = unreadMessages.take(2).toList();
+        fillWithMessages(unreadMessages + readMessages);
+
+        model.handleUpdateMessageEvent(eg.updateMessageEventMoveFrom(
+          origMessages: messagesToMove,
+          newTopicStr: newTopic));
+        checkNotifiedOnce();
+        checkMatchesMessages([
+          ...copyMessagesWith(messagesToMove, newTopic: newTopic),
+          ...unreadMessages.skip(2),
+        ]);
+      });
+
+      test('moving some read and unread messages from a conversation', () async {
+        await prepareStore();
+        final messagesToMove = [unreadMessages.first, readMessages.first];
+        fillWithMessages(unreadMessages + readMessages);
+
+        model.handleUpdateMessageEvent(eg.updateMessageEventMoveFrom(
+          origMessages: messagesToMove,
+          newTopicStr: newTopic));
+        checkNotifiedOnce();
+        checkMatchesMessages([
+          ...copyMessagesWith(unreadMessages.take(1), newTopic: newTopic),
+          ...unreadMessages.skip(1),
+        ]);
+      });
+
+      test('moving some read and all unread messages from a conversation', () async {
+        await prepareStore();
+        final messagesToMove = unreadMessages + readMessages.take(2).toList();
+        fillWithMessages(unreadMessages + readMessages);
+        final originalMessageIds =
+          model.streams[origChannel.streamId]![TopicName(origTopic)]!;
+
+        model.handleUpdateMessageEvent(eg.updateMessageEventMoveFrom(
+          origMessages: messagesToMove,
+          newTopicStr: newTopic));
+        checkNotifiedOnce();
+        checkMatchesMessages([
+          ...copyMessagesWith(unreadMessages, newTopic: newTopic),
+        ]);
+        final newMessageIds =
+          model.streams[origChannel.streamId]![TopicName(newTopic)]!;
+        check(originalMessageIds).identicalTo(newMessageIds);
+      });
+
+      test('moving all read and unread messages from a conversation', () async {
```

Comment:
This test seems redundant with the one above it, "moving some read and all unread messages from a conversation". So we can leave it out.

(The only difference is whether there are 2 or 10 read messages included in the list to move. In both cases there are 10 read messages included in `fillWithMessages` in setup; but read messages have no effect in `fillWithMessages`, so that doesn't matter.) 

I guess in particular this means that for one of the two tests I asked for at https://github.com/zulip/zulip-flutter/pull/1311#discussion_r1988053653 , all that was needed was a rename to make clearer that an existing test covered that scenario :slightly_smiling_face:

Reply:
Ha, I see. Yeah it occurred to me that the current test names aren't accurate enough about what they are doing, so it was no so clear whether they cover all the cases we care about.

Updated the names to express the relationships between the messages in terms of sets.